client/src/app/site/deployment-center/deployment-center-setup/step-configure/configure-vsts-build/functions-frameworks/functions-frameworks.component.ts

```diff
@@ -29,6 +30,21 @@ export class FunctionsFramworksComponent {
       value: WebAppFramework.PrecompiledFunction,
     },
   ];
+
+  private setupValidators() {
+    const workingDirectoryValidator = RegexValidator.create(
+      new RegExp(/^(\.{2,}|~|())\\(.)*$|^(\.{2,}|~|())\/(.)*$|(.)+:(.)*/),
+      this._translateService.instant(PortalResources.validate_workingDirectory),
+      true
+    );
+    this.wizard.buildSettings.get('applicationFramework').valueChanges.subscribe(stack => {
+      this.wizard.buildSettings.get('workingDirectory').setValidators([workingDirectoryValidator]);
+      this.wizard.buildSettings.get('workingDirectory').updateValueAndValidity();
+    });
+  }
+
   selectedFramework = WebAppFramework.ScriptFunction;
-  constructor(public wizard: DeploymentCenterStateManager, private _translateService: TranslateService) {}
+  constructor(public wizard: DeploymentCenterStateManager, private _translateService: TranslateService) {
```

Comment:
you can also create a removevalidator which can be called at the end. This might require html changes by adding the model and destructor function there.
This might help us in keeping the page light in future as subscriptions will increase

Reply:
Didn't get you, where do we call removevalidator? For functions we need the validation for all values in dropdown, so unlike other pages, I didn't add a removevalidator.. As for subscriptions from my understanding it's been taken care of by rxjs.. I'll implement the rxjs one.

Reply:
yeah, in my opinion both were needed as 
1. Subscriptions will be cleared when ngunsuscribe is called
2. Validations setup using portal, if not cleaned up by portal may cause problem later on. We can explicitly set them to null. (only added precaution)